tests/integration/test_index_routines.py

```diff
@@ -412,6 +427,25 @@ def test_axes_none(self):
         with pytest.raises(TypeError):
             num.diagonal(self.a, 0, None, 0)
 
+    def test_scalar_axes(self):
+        with pytest.raises(ValueError):
+            num.diagonal(self.a, axes=(0,))
+
+    def test_duplicate_axes(self):
+        with pytest.raises(ValueError):
+            num.diagonal(self.a, axis1=1, axes=(0, 1))
+        with pytest.raises(ValueError):
+            num.diagonal(self.a, axis1=1, axis2=0, axes=(0, 1))
+
+    def test_extra_axes(self):
+        axes = num.arange(self.a.ndim + 1, dtype=int)
+        with pytest.raises(ValueError):
+            num.diagonal(self.a, axes=axes)
+
+    def test_n_axes_offset(self):
+        with pytest.raises(ValueError):
+            num.diagonal(self.a, offset=1, axes=(2, 1, 0))
```

Comment:
For these three tests that only assert a cunumeric exception, what is the numpy behavior? If numpy doesn't raise the same error then these tests seem like they ought to be xfails instead, unless this is an intended divergence from numpy.

Reply:
NumPy doesn't have "axes" arg at all.

Reply:
@bryevdv any more comments on this one?

Reply:
> NumPy doesn't have "axes" arg at all.

In that case maybe just add a comment to that effect. We could also add `@pytest.mark.diff` that was introduced in another recent PR (just a way to quickly identify tests that codify an intended/expected difference from numpy)